pageserver/src/metrics.rs

```diff
@@ -1470,8 +1477,24 @@ impl SmgrOpFlushInProgress {
                     } else {
                         "slow flush completed or cancelled"
                     };
+
+                    let (inq, outq) = {
+                        // SAFETY: caller guarantees that `socket_fd` outlives this function.
```

Comment:
I'm quite convinced that's true, but what happens if we are wrong? Panic?
If that's the case we could ` fcntl(fd, F_GETFD) ` first, but that may lie if the fd is reused.

Reply:
If the fd number hasn't been reused yet the ioctl syscall fails with EBADFD and we log -2.

If the fd number gets reused for another tcp socket: we don't notice and print wrong data.

If the fd number gets reused for another resource from not-the-tcp subsystem (e.g. file on disk): behavior depends on whether that resource has an ioctl defined for the command number we're passing.
If we're lucky it doesn't implement ioctl at all or doesn't implement a command with that number.
In that case we get back EINVAL and we log -2.

Reply:
ioctl numbers are

```
#define [FIONREAD](https://elixir.bootlin.com/linux/v6.1.128/C/ident/FIONREAD)	0x541B
#define [TIOCOUTQ](https://elixir.bootlin.com/linux/v6.1.128/C/ident/TIOCOUTQ)	0x5411
```


I _think_ Linux tries to not reuse ioctl command nubmers for different subsystems

https://docs.kernel.org/userspace-api/ioctl/ioctl-number.html

So most likely we'll get EINVAL and log -2.

Reply:
Thanks! Seems low risk. lgtm